### PR TITLE
a far less wordy init flow

### PR DESCRIPTION
**Pull request by Aylr**
## What this (mostly) looks like

- Note there is a bit of post-production magic to make the welcome message appear faster.

render1573504495954

### CODE REVIEW

File: great_expectations/data_context/util.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def instantiate_class_from_config(config, runtime_config, config_defaults=None):
 
     class_name = config.pop("class_name", None)
     if class_name is None:
-        logger.warning("Instantiating class from config without an explicit class_name is dangerous. Consider adding "
```
Comment (jcampbell) [inline great_expectations/data_context/util.py]: I agree this should definitely not show up during init, but I believe this should indeed be a warning. It appears that the addition of the CTA config option didn't include a class_name, which is triggering the warning newly. I think this line stays the same, and `templates.py` in data context needs to change from:

```
...
    site_index_builder:
        show_cta_footer: True
```

to

```
...
    site_index_builder:
        class_name: DefaultSiteIndexBuilder
        show_cta_footer: True
```

**Review by jcampbell — CHANGES_REQUESTED** [review]
Looks awesome; one small change to address the warning about a missing class_name.

File: great_expectations/data_context/util.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def instantiate_class_from_config(config, runtime_config, config_defaults=None):
...
+        logger.debug("Instantiating class from config without an explicit class_name is dangerous. Consider adding "
```
Comment (jcampbell) [inline great_expectations/data_context/util.py]: A separate PR now adds the class_name default

```suggestion
        logger.warning("Instantiating class from config without an explicit class_name is dangerous. Consider adding "
```

**Review by jcampbell — CHANGES_REQUESTED** [review]
One line-in change now noted, since a separate PR addresses the issue that was causing the warning.

**Review by eugmandel — APPROVED** [review]

**Pull request merged**